File: Remove-Duplicates-from-Sorted-Array-II/Remove-Duplicates-from-Sorted-Array-II.py

class Solution:
    # @param A a list of integers
    # @return an integer
    def removeDuplicates(self, A):
        if len(A)==0:
            return 0
        pos=1
        num=1
        last=A[0]
        for i in range(1,len(A)):
            if A[i]!=last:
                A[pos]=A[i]
                pos+=1
                last=A[i]
                num=1
            elif num==2:
                continue
            else:
                num+=1
                A[pos]=last
                pos+=1
        return pos
 

File: Remove-Duplicates-from-Sorted-Array-II/Remove-Duplicates-from-Sorted-Array-II.cpp

class Solution {
public:
    int removeDuplicates(int A[], int n) {
        int ans=1;
        for(int i=1;i<n;i++)
            if(i==1 || A[i]!=A[ans-2])
                A[ans++]=A[i];
        return n==0?0:ans;
    }
};

 